branchLayout/api/src/main/java/com/virtuslab/branchlayout/api/IBranchLayoutEntry.java

```diff
@@ -14,36 +17,39 @@
 public interface IBranchLayoutEntry {
   String getName();
 
-  List<IBranchLayoutEntry> getChildren();
-
-  IBranchLayoutEntry withChildren(List<IBranchLayoutEntry> newChildren);
-
   @Nullable
   String getCustomAnnotation();
 
-  default boolean equals(IBranchLayoutEntry other) {
-    val areNamesSame = this.getName().equals(other.getName());
-    val thisCustomAnnotation = this.getCustomAnnotation();
-    val otherCustomAnntation = other.getCustomAnnotation();
+  List<IBranchLayoutEntry> getChildren();
 
-    var areCustomAnnotationsSame = thisCustomAnnotation == null && otherCustomAnntation == null;
+  IBranchLayoutEntry withChildren(List<IBranchLayoutEntry> newChildren);
 
-    if (thisCustomAnnotation != null && otherCustomAnntation != null) {
-      areCustomAnnotationsSame = thisCustomAnnotation.equals(otherCustomAnntation);
-    }
+  @EnsuresNonNullIf(expression = "#2", result = true)
+  static boolean defaultEquals(@FindDistinct IBranchLayoutEntry self, @Nullable Object other) {
+    if (self == other) {
+      return true;
+    } else if (!(other instanceof IBranchLayoutEntry)) {
+      return false;
+    } else {
+      val otherEntry = (IBranchLayoutEntry) other;
+      val areNamesSame = self.getName().equals(otherEntry.getName());
+      val areCustomAnnotationsSame = Objects.equals(self.getCustomAnnotation(), otherEntry.getCustomAnnotation());
 
-    if (areNamesSame && areCustomAnnotationsSame && this.getChildren().size() == other.getChildren().size()) {
-      val entryNameComparator = Comparator.comparing(IBranchLayoutEntry::getName);
+      if (areNamesSame && areCustomAnnotationsSame && self.getChildren().size() == otherEntry.getChildren().size()) {
+        val entryNameComparator = Comparator.comparing(IBranchLayoutEntry::getName);
 
-      val sortedThisEntries = this.getChildren()
-          .sorted(entryNameComparator);
-      val sortedOtherEntries = other.getChildren()
-          .sorted(entryNameComparator);
+        val sortedSelfEntries = self.getChildren().sorted(entryNameComparator);
+        val sortedOtherEntries = otherEntry.getChildren().sorted(entryNameComparator);
 
-      return sortedThisEntries.zip(sortedOtherEntries)
-          .forAll(entryTuple -> entryTuple._1.equals(entryTuple._2));
+        return sortedSelfEntries.zip(sortedOtherEntries)
+            .forAll(entryTuple -> entryTuple._1.equals(entryTuple._2));
```

Comment:
`.equals` or `defaultEquals`?

Reply:
`defaultEquals` is just method implementation, to be shared by all classes implementing the interface. In fact I'd make it just `default boolean equals(Object other)` in the interface... but Java doesn't allow defining implementing `equals`/`hashCode`/`toString` in interfaces :/

Reply:
So `.equals` is to be used everywhere